### PR TITLE
Update Swiftmailer configuration docs

### PR DESCRIPTION
**Pull request by jverdeyen**
Added `delivery_addresses` configuration.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@jverdeyen thanks for documenting this missing option. Looking at the code of Swiftmailer, we can see the following:

```
The swiftmailer.delivery_address configuration key is deprecated since version 2.3.10
and will be removed in 3.0. Use the swiftmailer.delivery_addresses configuration key
instead (or remove the empty setting).
```

So I guess we should also do the following:

* Remove the description of the `delivery_address` option in 3.* docs
* Add a warning message in 3.* docs telling that `_addresses` was previously called `_address`
* Remove in the Symfony code this `_address` option

**Comment by jverdeyen** [comment]
Updated!

**Comment by javiereguiluz** [comment]
I was going to create a new issue in the Symfony repo to mention the `delivery_address` deprecation ... but then I remembered that SwiftMailerBundle is one of the "built-in" bundles that is not really built-in ... so the 2.3.10 and 3.0 versions mentioned by the deprecation refer to SwiftMailerBundle versions and not Symfony versions.

If that's correct, my previous comment was completely wrong and we should revert the b34d046 commit. I'm really sorry.

**Comment by jverdeyen** [comment]
So both `delivery_address` and `delivery_addresses` are valid in 3.* ?

**Comment by javiereguiluz** [comment]
@jverdeyen yes, because even the master branch in the Symfony Standard edition requires this:

```json
"symfony/swiftmailer-bundle": "^2.3",
```

See https://github.com/symfony/symfony-standard/blob/master/composer.json#L19

**Comment by jverdeyen** [comment]
@javiereguiluz Updated!

**Comment by javiereguiluz** [comment]
👍 

@jverdeyen thanks for completing this and sorry for having given you wrong directions in my previous comments.

**Comment by jverdeyen** [comment]
@javiereguiluz No problem!

File: email/dev_environment.rst [hunk 1]
```diff
@@ -53,12 +53,12 @@ will not be sent when you run tests, but will continue to be sent in the
 If you'd also like to disable deliver in the ``dev`` environment, simply
 add this same configuration to the ``config_dev.yml`` file.
 
-Sending to a Specified Address
+Sending to a Specified Address(es)
 ------------------------------
```
Comment (stof) [inline email/dev_environment.rst]: you need to make the underlining longer here as the title is longer

**Comment by stof** [comment]
@javiereguiluz but the policy was to avoid documenting deprecated features though, so the doc should describe only the new way IMO

**Comment by javiereguiluz** [comment]
@jverdeyen I've updated your pull request to remove any mention to the deprecated `delivery_address` option. I did this myself because I don't want to make you waste more time on this :)

I have a question for XML config experts (@xabbug, @wouterj)  How can I define the `delivery_addresses` config as an array in XML? Thanks!

**Comment by xabbuh** [comment]
@javiereguiluz You wil define several `delivery-address` element nodes (see https://github.com/symfony/swiftmailer-bundle/blob/master/Resources/config/schema/swiftmailer-1.0.xsd#L17)

File: reference/configuration/swiftmailer.rst [hunk 2]
```diff
@@ -235,7 +236,6 @@ Full Default Configuration
                 auth_mode=""
                 sender_address=""
                 delivery_address=""
```
Comment (xabbuh) [inline reference/configuration/swiftmailer.rst]: done

**Comment by jverdeyen** [comment]
Any update?

**Comment by xabbuh** [comment]
Thank you @jverdeyen.

**Pull request closed without merging**